src/mono/browser/runtime/marshal-to-js.ts

```diff
@@ -338,6 +339,10 @@ function create_task_holder(res_converter?: MarshalerToJs) {
 }
 
 export function mono_wasm_resolve_or_reject_promise(args: JSMarshalerArguments): void {
+    // rejection/resolution should not arrive earlier than the promise created by marshaling in mono_wasm_invoke_jsimport_MT 
+    invoke_later_when_on_ui_thread_async(() => mono_wasm_resolve_or_reject_promise_impl(args));
+}
+export function mono_wasm_resolve_or_reject_promise_impl(args: JSMarshalerArguments): void {
```

Comment:
nit: Same name pattern as above?

```suggestion
export function mono_wasm_resolve_or_reject_promise_ui(args: JSMarshalerArguments): void {
```

Reply:
mono_wasm_resolve_or_reject_promise_impl is also called on WebWorker ... so no rename